capsules/extra/src/hmac.rs

```diff
@@ -147,9 +147,7 @@ impl<
                             })
                         })
                         .unwrap_or(Err(ErrorCode::RESERVE));
-                    if ret.is_err() {
-                        return ret;
-                    }
+                    ret?;
```

Comment:
I know this is super-Rusty, but I need my IDE to quintuple the font size of the `?` or something — hiding an early-exit from a function in a single character is not my favorite design decision.

Reply:
Yeah, that's the pain point with not having these lints on from the start. The `?` should be on the original function call. But then that changes the types and this is no longer a mechanical change, it's more invasive.

If the developer had gotten this clippy warning, they would have realized they could have used the `?` earlier and we wouldn't have this odd line. This is fixable, but not in the spirit of these clippy prs.